### ISSUE TITLE
Paying for Sunburst

### ISSUE THREAD

**Issue by dwetuski** [Card-Forge/forge#3009]
**Is your feature request related to a problem? Please describe.**
When tapping lands that allow choosing a color to pay for a creature spell with sunburst, Forge arbitrarily picks colors to use as colorless mana, often leading to less-than-optimum results.

**Describe the solution you'd like**
I'd like Forge to always prompt for which color mana to use when tapping a source that offers such a choice.

**Describe alternatives you've considered**
In cases where only one option has not already been used, Forge could automatically pick the previously unpicked color to maximize the number of +1/+1 counters.  However, there may be times when having a smaller creature is preferable, e.g. if Ensnaring Bridge is on the battlefield.

**Additional context**
Currently, I work around this issue by tapping for mana before clicking the spell to cast.  However, this isn't always an option, e.g. when a sunburst card is selected by cascade.  Another workaround is using Dev Mode to add counters, but this too doesn't always work, because there may be global toughness-reducing effects that could cause the creature to die if it lacks sufficient +1/+1 counters.

**Comment by dwetuski** [comment]
Right, so, what you're trying to do is a little different from what I was talking about, but I've attached a game state where the auto-pay picked the same color mana and only gave 1 +1/+1 counter on a {2} cost sunburst card.
[sunburst.txt](https://github.com/Card-Forge/forge/files/11821933/sunburst.txt)

Also, the AI seems to prefer colorless lands over any color creatures as mana sources when auto-paying a sunburst cost (it prefers Reliquary Tower over Alloy Myr), but this is understandable behavior, even if it's not always desirable behavior.

Incidentally, the behavior I described when originally submitting this issue seems to be fixed in version 1.6.56, so thank you to whoever did that.

**Comment by Agetian** [comment]
Ah, I see that in that game state above, the Sunburst is not even inherent to the card, but is gained from a different source when you play the spell - that's what tricks the AI into misunderstanding that it'll even have Sunburst in the first place. At any rate, yes, the mana choice was fixed/improved recently, so in general it should be better in simpler cases (like casting a spell that inherently has Sunburst), but corner cases like the one you've shown are likely to still be problematic without manual payment. Not sure, but this might be tricky-ish to improve, will need to think on this one.
